### ISSUE TITLE
New commands "stamp list" and "stamp retrieve"

### ISSUE THREAD

**Issue by euneuber** [pdfcpu/pdfcpu#302]
Thank you for this great package!

How would someone list all annotations (stamps) for a given PDF?
Maybe an extension to the stamp command would be useful (especially for work flows!) so an annotation can also read an processed...

**Comment by hhrutter** [comment]
Hello!

Some clarification upfront:

pdfcpu's stamp facility does not generate watermark page annotations like described in the PDF spec.

In a nutshell `pdfcpu stamp add` just tries to replicate Adobe's stamp facility by packing your stamp data into a form and then appending the form to the page content stream.

Once a page is stamped you can either `pdfcpu stamp remove` or `pdfcpu stamp update` this one logical page stamp.

You can `add` a stamp anytime but that really treats an already stamped page just as another source page to stamp.
So you can stamp on top each time but you lose context and the ability to roll back all your stamps after the fact in case
you wanted to keep track of your individual stamps.

TL;DR: `pdfcpu stamp` is  just a naive unorganized one-off stamping for any given PDF Page.
The idea is to eventually build in some organization of stamps (by giving them ids and such) on a page basis using logical layers for editing oder removing specific stamps later on.

-----------------------

Given this context I am not sure how to answer your question.
Please rephrase your request and in any case please describe your specific use case.
Maybe we can work out some valuable improvements for the stamping facility.

Thank you for using pdfcpu 💚

**Comment by euneuber** [comment]
How would you know that a very stamp is contained in a PDF?
I'd like to query the annotations/stamps within the software package that created it.
What do you think?

**Comment by hhrutter** [comment]
Hello!

I am sorry I was unclear.
A pdfcpu stamp/watermark is similar to the Adobe watermark tool and has nothing to do with Adobe stamp comments
that you can move around.
https://helpx.adobe.com/acrobat/using/add-watermarks-pdfs.html
https://helpx.adobe.com/acrobat/using/adding-stamp-pdf.html

Both of the following commands
https://pdfcpu.io/core/stamp
https://pdfcpu.io/core/watermark
generates elements at a fixed position on any number of selected pages.
As you apply a sequence of stamps/watermarks consecutively to a page they immediately become part of the page content.

At this point pdfcpu is not dealing with annotations (like the stamp annotation) which is used by Adobe behind the scenes of stamp comments.

In order to better understand your use case do you think you can share a sample PDF and use it to describe your issue?
It's easy to mix up the proper terms, you can also hit me up in the #pdfcpu slack channel.

**Comment by hhrutter** [comment]
v0.3.12 allows you to list and remove annotations.
Maybe this is smth useful for you.
